File: HelloWorld-Csharp/HelloWorld/test_HelloWorld/test_SayHelloWorld.cs
using System;
using HelloWorld;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace test_HelloWorld
{
    [TestClass]
    public class Test_SayHelloWorld
    {
        [TestMethod]
        public void TestInteger()
        {

            String[] testArgs = { "5", "2" };

            Assert.AreEqual(5, HelloWorld.HelloWorld.SayHelloWorld(testArgs));
        }

        [TestMethod]
        public void TestZero()
        {

            String[] testArgs = { "0", "2" };

            Assert.AreEqual(0, HelloWorld.HelloWorld.SayHelloWorld(testArgs));
        }

        [TestMethod]
        public void TestFloat()
        {

            String[] testArgs = { "8.45", "2" };

            Assert.AreEqual(3, HelloWorld.HelloWorld.SayHelloWorld(testArgs));
        }

        [TestMethod]
        public void TestNegative()
        {

            String[] testArgs = { "-7", "2" };

            Assert.AreEqual(3, HelloWorld.HelloWorld.SayHelloWorld(testArgs));
        }

        [TestMethod]
        public void TestAlpha()
        {

            String[] testArgs = { "b", "2" };

            Assert.AreEqual(3, HelloWorld.HelloWorld.SayHelloWorld(testArgs));
        }

        [TestMethod]
        public void TestNoArgs()
        {

            // String[] testArgs = { null };
            String[] testArgs = null;

            Assert.AreEqual(3, HelloWorld.HelloWorld.SayHelloWorld(testArgs));
        }
    }
}

File: HelloWorld-Java/test/test_HelloWorld/test_SayHelloWorld.java
package test_HelloWorld;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

import HelloWorld.HelloWorld;

public class test_SayHelloWorld {

    @Test
    public void testInteger() {

	String[] testArgs = { "5", "2" };

	// HelloWorld myTest = new HelloWorld();
	assertEquals(5, HelloWorld.SayHelloWorld(testArgs));
    }

    @Test
    public void testZero() {

	String[] testArgs = { "0", "2" };

	// HelloWorld myTest = new HelloWorld();
	assertEquals(0, HelloWorld.SayHelloWorld(testArgs));
    }

    @Test
    public void testFloat() {

	String[] testArgs = { "8.45", "2" };

	// HelloWorld myTest = new HelloWorld();
	assertEquals(3, HelloWorld.SayHelloWorld(testArgs));
    }

    @Test
    public void testNegative() {

	String[] testArgs = { "-7", "2" };

	// HelloWorld myTest = new HelloWorld();
	assertEquals(3, HelloWorld.SayHelloWorld(testArgs));
    }

    @Test
    public void testAlpha() {

	String[] testArgs = { "b", "2" };

	// HelloWorld myTest = new HelloWorld();
	assertEquals(3, HelloWorld.SayHelloWorld(testArgs));
    }

    @Test
    public void testNoArgs() {

	// String[] testArgs = { null };
	String[] testArgs = null;

	// HelloWorld myTest = new HelloWorld();
	assertEquals(3, HelloWorld.SayHelloWorld(testArgs));
    }

}
